python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -165,6 +171,15 @@ def _step_output_error_checked_user_event_sequence(
                     f'Emitting implicit Nothing for output "{step_output_def.name}" on {op_label}'
                 )
                 yield Output(output_name=step_output_def.name, value=None)
+
+            if step_context.is_sda_step and step_context.get_observed_user_asset_mat(
```

Comment:
Not sure if this has any consequence with the way that execution currently works, but imagine a world where we could kick off downstream steps as soon as an upstream output is produced, even if the upstream step hasn't finished yet.

In that world, we would ideally yield the `Output` as soon the materialization is reported/yielded/returned, right?